keps/sig-storage/3476-volume-group/README.md

```diff
@@ -0,0 +1,1680 @@
+
+## Proposal for Volume Placement
+
+### API Changes
+
+In order to support Volume Placement, An `AllowedTopologies` field will be added to the VolumeGroupClass API:
+
+```
+type VolumeGroupClass struct {
+        metav1.TypeMeta
+        // +optional
+        metav1.ObjectMeta
+
+        // Driver is the driver expected to handle this VolumeGroupClass.
+        // This value may not be empty.
+        Driver string
+
+        // Parameters holds parameters for driver.
+        // These values are opaque to the system and are passed directly
+        // to the driver.
+        // +optional
+        Parameters map[string]string
+
+        // This field specifies whether group snapshot is supported.
+        // The default is false.
+        // +optional
+        VolumeGroupSnapshot *bool
+
+        // Restrict the topologies where a group of volumes can be located.
+        // Each driver defines its own supported topology specifications.
+        // An empty TopologySelectorTerm list means there is no topology restriction.
+        // This field is passed on to the drivers to handle placement of a group of
+        // volumes on storage pools.
+        // +optional
+        AllowedTopologies []api.TopologySelectorTerm
+}
+```
+
+#### ModifyVolume
+
+ModifyVolume CSI RPC was considered earlier to add/remove one volume to/from a group at a time but it was removed because ModifyVolumeGroup CSI RPC was added.
+
+A new MODIFY_VOLUME capability will be added to support this.
+It indicates that the controller plugin supports modifying a volume.
+
+```
+  rpc ModifyVolume(ModifyVolumeRequest)
+    returns (ModifyVolumeResponse) {
+        option (alpha_method) = true;
+    }
+```
+
+This RPC is called when an existing volume is added to an existing volume group or when a volume is removed from the volume group.
+A volume group id parameter will be in the ModifyVolumeRequest for an add request.
+A volume group id parameter will not be in the ModifyVolumeRequest for a delete request.
+If user requests to add an existing volume to a consistency group, but the CSI driver cannot fulfill the request because the existing volume is placed on a different storage pool from the consistency group, then the CSI driver MUST return failure.
+This RPC MUST be idempotent.
+
+```
+message ModifyVolumeRequest {
+      string volume_id = 1;
+
+      // This field is OPTIONAL.
+      repeated string volume_group_id = 2 [(alpha_field) = true];
+
+      // Secrets required by plugin to complete modify volume request.
+      // This field is OPTIONAL. Refer to the `Secrets Requirements`
+      // section on how to use this field.
+      map<string, string> secrets = 3 [(csi_secret) = true];
+}
+```
+External-provisioner will be modified so that modifying PVC by adding VolumeGroupName will trigger a ModifyVolume call (a new CSI controller RPC) to CSI driver.
+
+#### Create VolumeGroup with Selector
```

Comment:
We do support selector now? Can this be removed?

Reply:
Create VolumeGroup from existing PVCs with a selector is moved to Phase 2 as I want to make the scope of this KEP smaller initially.
However, if we are not adding a new field in PVC, we could apply a label with the group name to a PVC to trigger adding an existing PVC to a group.